acceptance_tests/features/steps/display_respondent_alongside_response_status.py

```diff
@@ -0,0 +1,115 @@
+import logging
+
+import time
+from behave import given, when, then
+from structlog import wrap_logger
+
+from acceptance_tests import browser
+from acceptance_tests.features.pages import reporting_unit, collection_exercise_details, \
+    change_response_status, surveys_todo
+from acceptance_tests.features.pages.surveys_history import go_to_history_tab
+from acceptance_tests.features.steps.authentication import signed_in_internal, signed_in_respondent
+from acceptance_tests.features.steps.change_response_status import go_to_reporting_unit_page
+from controllers.collection_exercise_controller import map_ce_status
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+@given('a user has completed a survey online')
+def enrolling_onto_survey(context):
+    signed_in_respondent(context)
+    browser.click_link_by_id('access_survey_button_1')
+    collection_exercise_details.upload_collection_instrument(
+        test_file='resources/collection_instrument_files/064_201803_0001.xlsx')
+    browser.find_by_id('upload_survey_button').click()
+
+
+@given('an external user is enrolled onto a given survey')
+def enrolled_respondent_but_not_completed(context):
+    signed_in_respondent(context)
+
+
+@given('a user has completed a survey via the telephone')
+def user_completes_survey_by_telephone(context):
+    signed_in_internal(())
+    go_to_reporting_unit_page(context)
+    reporting_unit.click_data_panel(context.short_name)
+    reporting_unit.click_change_response_status_link(survey=context.short_name, period=context.period)
+    mapped_status = map_ce_status('Completed by phone')
+    change_response_status.update_response_status(mapped_status)
+
+
+@given('the status is set to "{status}"')
+def survey_set_to_longer_required(context, status):
+    signed_in_respondent(context)
+    go_to_history_tab()
+    assert browser.find_by_text(status)
+
+
+@given('the status is set to In Progress')
+def survey_set_to_in_progress(_):
+    surveys_todo.go_to()
+    assert browser.find_by_text('Downloaded')
+
+
+@given('an internal user has set a given survey to "No longer required" for a given respondent')
+def survey_is_set_to_no_longer_required(context):
+    signed_in_internal(())
+    go_to_reporting_unit_page(context)
+    reporting_unit.click_data_panel(context.short_name)
+    reporting_unit.click_change_response_status_link(survey=context.short_name, period=context.period)
+    mapped_status = map_ce_status('No longer required')
+    change_response_status.update_response_status(mapped_status)
+
+
+@given('a user has partially completed a survey (In progress)')
+def respondent_partially_completes_survey(context):
+    signed_in_respondent(context)
+    browser.click_link_by_id('access_survey_button_1')
+    browser.click_link_by_id('download_survey_button')
+    time.sleep(5)
```

Comment:
I suppose we could do something more elegant than this

Reply:
I'm trying to see if I can get rid of the wait, the only worry is that it attempts to go to the surveys page with the window that closes once it downloads.